### PR TITLE
1158 hide personal fields

### PR DESCRIPTION
**Pull request by leventegal-she**

### CODE REVIEW

File: sormas-api/src/main/java/de/symeda/sormas/api/caze/CaseJurisdictionDto.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	private static final long serialVersionUID = -5412823431238056752L;
+
+	private String reportingUserUuid;
+	private String regionUui;
+	private String districtUud;
+	private String communityUuid;
+	private String healthFacilityUuid;
+	private String pointOfEntryUuid;
+
+	public CaseJurisdictionDto() {
```
[the remark this answers is not in the document]
Comment (leventegal-she) [reply sormas-api/src/main/java/de/symeda/sormas/api/caze/CaseJurisdictionDto.java:34]: It is needed when for creating CaseJurisdictionDto for a `Case` see `JurisdictionHelper`.

File: sormas-api/src/main/java/de/symeda/sormas/api/sample/SampleExportDto.java [hunk 2]
```diff
@@ -27,8 +32,7 @@
 	private String uuid;
 	private String labSampleID;
 	private String epidNumber;
-	private String firstName;
-	private String lastName;
+	private AssociatedCase associatedCase;
```
[the remark this answers is not in the document]
Comment (leventegal-she) [reply sormas-api/src/main/java/de/symeda/sormas/api/sample/SampleExportDto.java:35]: It is necessary because after implementing export samples with contacts the fistName and lastName would come from the contact person and it would be a bit messy to have case and contact data in separate objects but their person name in common fields

File: sormas-backend/src/main/java/de/symeda/sormas/backend/util/PseudonymizationService.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+	private FieldAccessCheckers createFieldAccessCheckers(boolean isInJurisdiction) {
+		return new FieldAccessCheckers()
+				.add(new PersonalDataFieldAccessChecker(r -> userService.hasRight(r), isInJurisdiction));
+	}
+
+	private <DTO> void pseudonymizeField(DTO dto, Field field) {
+		try {
+			Object emptyValue = field.getType().equals(String.class) ? "" : null;
+
+			field.setAccessible(true);
```
[the remark this answers is not in the document]
Comment (leventegal-she) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/util/PseudonymizationService.java:107]: I've never made reset, but you're right it could be reset :)

File: sormas-backend/src/test/java/de/symeda/sormas/backend/person/PersonFacadeEjbPseudonymizationTest.java [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+
+	@Test
+	public void testGetEventParticipantPersonInOtherJurisdiction() {
+		EventDto event = creator.createEvent(user1.toReference());
+		creator.createEventParticipant(event.toReference(), person);
+
+		assertPsoudonimysed(person);
+	}
+
+	private PersonDto createPerson() {
```
[the remark this answers is not in the document]
Comment (leventegal-she) [reply sormas-backend/src/test/java/de/symeda/sormas/backend/person/PersonFacadeEjbPseudonymizationTest.java:218]: This method is specific to this test I've already added a method which accepts and saves the address of the person. I don't think it's a good idea to have hardcoded values in TestEntityCreator.

File: sormas-app/app/build.gradle [hunk 5]
```diff
@@ -91,7 +91,7 @@ dependencies {
     implementation 'com.google.firebase:firebase-config:19.1.1'
     implementation 'com.google.firebase:firebase-analytics:17.2.2'
     implementation 'com.crashlytics.sdk.android:crashlytics:2.10.1'
-    implementation "de.symeda.sormas:sormas-api:$sormasVersion"
+    implementation("de.symeda.sormas:sormas-api:$sormasVersion") { changing = true }
```
[the remark this answers is not in the document]
Comment (leventegal-she) [reply sormas-app/app/build.gradle:94]: After updating and installing the project Android Studio doesn't reload the changes from sormas-api jar, the app code compiles but the IDE shows the new things in red. 
This change does a kind of force to refresh this dependency in Android Studio.

[the remark this answers is not in the document]
Comment (MateStrysewskeSym) [reply sormas-app/app/build.gradle:94]: Strange, didn't happen to me so far, but alright!

File: sormas-backend/src/test/java/de/symeda/sormas/backend/person/PersonFacadeEjbPseudonymizationTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (MateStrysewskeSym) [reply sormas-backend/src/test/java/de/symeda/sormas/backend/person/PersonFacadeEjbPseudonymizationTest.java:218]: Okay!

File: sormas-api/src/main/java/de/symeda/sormas/api/sample/SampleExportDto.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (MateStrysewskeSym) [reply sormas-api/src/main/java/de/symeda/sormas/api/sample/SampleExportDto.java:35]: Okay, I'm fine leaving it like this for now.

File: sormas-ui/src/main/java/de/symeda/sormas/ui/utils/AbstractFilterForm.java [hunk 6]
```diff
@@ -167,7 +167,7 @@ private void onFieldValueChange(String propertyId, Property.ValueChangeEvent eve
 				this.getFieldGroup().commit();
 				this.fireValueChange(true);
 			} catch (FieldGroup.CommitException ex) {
-				ex.printStackTrace();
+				// do nothing
```
Comment (MateStrysewskeSym) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/utils/AbstractFilterForm.java:170]: Sorry, I've just seen that the printStackTrace is already on the development branch, i.e. it's intended. I don't know why I thought that you added it. You can revert this change.
